tests/pkg/k8sclient/cspc_util.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8sclient
+
+import (
+	. "github.com/onsi/gomega"
+	cstor "github.com/openebs/api/pkg/apis/cstor/v1"
+	"github.com/openebs/api/pkg/apis/openebs.io/v1alpha1"
+	"github.com/openebs/api/pkg/apis/types"
+	"k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+const maxRetry = 30
+
+// GetHealthyCSPICountEventually gets online cspi(s) based on cspc name
+func (client *Client) GetOnlineCSPICountEventually(cspcName, cspcNamespace string, expectedCSPICount int) int {
```

Comment:
Instead of having `n` different functions like `GetOnlineCSPICountEventually`, `GetCSPICountEventually` and `GetDegradedCSPICountEventually` we can have single function `GetCSPICountEventually` and take list of Predicates as input along with cspcName, cspcNamespace and expectedCSPICount.

Reply:
Yep... Will take it in the next PR.